opm/material/Constants.hpp

```diff
@@ -74,6 +84,126 @@ class Constants
      */
     static const Scalar hRed;
 };
+template<>
+class Constants<float>
+{
+public:
+    /*!
+     * \brief The ideal gas constant [J/(mol K)]
+     */
+    static constexpr float R = 8.314472;
+
+    /*!
+     * \brief The Avogadro constant [1/mol]
+     */
+    static constexpr float Na = 6.02214179e23;
+
+    /*!
+     * \brief The Boltzmann constant [J/K]
+     */
+    static constexpr float kb = R/Na;
+
+    /*!
+     * \brief Speed of light in vacuum [m/s]
+     */
+    static constexpr float c = 299792458.0;
+
+    /*!
+     * \brief Newtonian constant of gravitation [m^3/(kg s^2)]
+     */
+    static constexpr float G = 6.67428e-11;
+
+    /*!
+     * \brief Planck constant [J s]
+     */
+    static constexpr float h = 6.62606896e-34;
+
+    /*!
+     * \brief Reduced Planck constant [J s]
+     */
+    static constexpr float hRed = h / (2 * M_PI);
+};
+template<>
+class Constants<double>
+{
+public:
+    /*!
+     * \brief The ideal gas constant [J/(mol K)]
+     */
+    static constexpr double R = 8.314472;
+
+    /*!
+     * \brief The Avogadro constant [1/mol]
+     */
+    static constexpr double Na = 6.02214179e23;
+
+    /*!
+     * \brief The Boltzmann constant [J/K]
+     */
+    static constexpr double kb = R/Na;
+
+    /*!
+     * \brief Speed of light in vacuum [m/s]
+     */
+    static constexpr double c = 299792458.0;
+
+    /*!
+     * \brief Newtonian constant of gravitation [m^3/(kg s^2)]
+     */
+    static constexpr double G = 6.67428e-11;
+
+    /*!
+     * \brief Planck constant [J s]
+     */
+    static constexpr double h = 6.62606896e-34;
+
+    /*!
+     * \brief Reduced Planck constant [J s]
+     */
+    static constexpr double hRed = h / (2 * M_PI);
```

Comment:
Note: Very, very, very strictly speaking there's no such thing as `M_PI`.  It's a common Posix extension, but it does not exist in portable C++.  I'm not going to request that it be removed however, because we get [`std::numbers::pi`](https://en.cppreference.com/w/cpp/numeric/constants) in C++20.